README.md

```diff
@@ -223,6 +237,22 @@ Example of customizing semantic colors in settings.json:
 
     -   Convert import format according to `python.analysis.importFormat`.
 
+# Auto/Add import and indexing
+
+Pylance has a feature called ```auto import``` which will show symbols from installed packages and user files in completion. When committed, insert ```import``` statement for you. It also has a similar feature called ```add import``` that will show symbols from installed packages and user files for an unknown identifier. When selected, insert ```import``` statement for you.
+
+Since these features can be costly to scan symbol information and show them in completion and code actions, by default, we have minimal functionality enabled and provide multiple settings to control its behavior.
+
+When ```python.analysis.indexing``` is off, we will only show symbols already used by opened files transitively. If ```indexing``` is on, we will spend some time indexing installed packages and user files when pylance is started, but after that, we will provide symbols that you haven't used yet in completion and code action.
+
+But since indexing is an expensive operation, we do not index every symbol from all installed packages but only top-level symbols. In another words, symbols in ```__all__``` in ```package/__init__```. If users want more than that, they need to use ```python.analysis.packageIndexDepths``` to tell us which package they want us to spend more time on. Symbols indexed will show up both in completion and code actions.
+
+For user files, we will index up to 2000 files. One difference between user files and installed packages is that they won't appear in completion unless it is already used. Unlike installed packages, we, by default, index all symbols from user files, which can cause thousands of user symbols to show up in completion. 
+
+We choose to show them only in code action since it requires explicit user action ```Ctrl+.``` rather than automatically triggered like completion.
```

Comment:
What is the 'them' you're referring to in this paragraph? Completions?

Reply:
by them, I meant symbols from user files.